samples/ProjectGrandmaster.Unity/Assets/Script/Board/BoardInformation.cs

```diff
@@ -0,0 +1,633 @@
+            EndGameResult.SetActive(false);
+        }
+
+
+        /// <summary>
+        /// Checkmate. Shows Checkmate to notify the players
+        /// </summary>
+        public void Checkmate(int colour)
+        {
+            string checkmate = "CHECKMATE";
+            SetText(checkmate);
+            EndGame(colour);
+        }
+
+        /// <summary>
+        /// Changes the game state
+        /// Declares the winner
+        /// </summary>
+        /// <param name="colour"> Colour that won. 0 if White. 1 if Black. -1 if Draw. </param>
+        public void EndGame(int colour)
+        {
+            GameEnded = true;
+            // White won
+            if (colour == 0) { Winner = 1; }
+
+            // Black won
+            else if (colour == 1) { Winner = -1; }
+
+            // Draw
+            else { Winner = 0; }
+
+            // Display result to players
+            ShowResult();
+            StartCoroutine(FlashText(5, false));
+        }
+
+        /// <summary>
+        /// Forfeit display
+        /// </summary>
+        /// <param name="colour">The colour of the side that forfeited.</param>
+        public void Forfeited(int colour)
+        {
+            GameEnded = true;
+            string text;
+            
+            // White won
+            if (colour == 1)
+            {
+                text = "Black forfeited!";
+                Winner = 1;
+            }
+
+            // Black won
+            else
+            {
+                text = "White forfeited!";
+                Winner = -1;
+            }
+
+            // Display result to players
+            SetText(text);
+            ShowResult();
+            StartCoroutine(FlashText(5, false));
+        }
+
+        /// <summary>
+        /// Displays the game result on top of the board
+        /// </summary>
+        /// <returns></returns>
+        private void ShowResult()
+        {
+            string won = "YOU WON!";
+            string lose = "YOU LOST :(";
+            string draw = "YOU DREW";
+
+            // Draw
+            if (Winner == 0)
+            {
+                whiteText.GetComponent<Text>().text = draw;
+                blackText.GetComponent<Text>().text = draw;
+            }
+            // White won
+            else if (Winner == 1)
+            {
+                whiteText.GetComponent<Text>().text = won;
+                blackText.GetComponent<Text>().text = lose;
+            }
+            // Black won
+            else
+            {
+                whiteText.GetComponent<Text>().text = lose;
+                blackText.GetComponent<Text>().text = won;
+            }
+        }
+
+        /// <summary>
+        /// Called when the pawn reaches the opposite side of the board.
+        /// Pawn disappears and player has a chance to choose the piece they want to replace it with.
+        /// </summary>
+        public IEnumerator PromotePawn(PieceInformation pieceInfo, GameObject pawn)
```

Comment:
Generally we pass around the specific component (like PieceInformation) and use "pieceInformation.gameObject", and it seems you are doing so here, but also passing in the "pawn" game object, why both?